addons/website/static/src/client_actions/website_preview/website_preview.js

```diff
@@ -338,7 +338,33 @@ export class WebsitePreview extends Component {
         this.title.setParts({ action: currentTitle });
     }
 
-    _onPageLoaded() {
+    _onPageLoaded(ev) {
+        // FIX Chrome-only. If you have the backend in a language A but the
+        // website in English only, you can 1) modify a record's (event,
+        // product...) name in language A (say "New Name").
+        // 2) visit the page `/new-name-11` => the server will redirect you to
+        // the English page `/origin-11`, which is the only one existing.
+        // Chrome caches the redirection.
+        // 3) give the same name in English as in language A, try to visit
+        // => the server now wants to access `/new-name-11`
+        // => Chrome uses the cache to redirect `/new-name-11` to `/origin-11`,
+        // => the server tries to redirect to `/new-name-11` => start again 🔄.
+        // Chrome injects a "Too many redirects" layout in the iframe, which in
+        // turn raises a CORS error when the app tries to update the iframe.
+        // If we detect that behavior, we reload the iframe with a new query
+        // parameter, so that it's not cached for Chrome.
+        if (navigator.userAgent.toLowerCase().includes("chrome") && !this.iframeReloaded) {
+            try {
+                this.iframe.el.contentWindow.location.href;
+            } catch (err) {
```

Comment:
We are inside `_onPageLoaded`, so it means that once the iframe has been loaded, you actually reload it in a try/catch?
Does that mean that on chrome, any iframe load will be done twice?
How does this new reload (when going fine, no error) not go into infinite loop? This reloads the iframe, which enters `_onPageLoaded` again no? Without `iframeReloaded` being set to short-circuit has everything went fine.

Can't we use something like `iframe.contentWindow.onerror`?

Reply:
There's no infinite loop because 1) if the `try` succeeds, we don't reload the iframe (so no, we don't reload twice every iframe load on Chrome), 2) if there's another error than a `SecurityError`, we don't reload, and 3) if it's a `SecurityError` that's caused by some other reason than this redirect, the `iframeReloaded` parameter (or as you noted, the search param in the URL) is checked. But that 3rd point is really mainly a safety belt.

I can test something with `iframe.contentWindow.onerror` though. That'd probably be prettier.

Reply:
Well, trying an `iframe.el.contentWindow.addEventListener("error", () => debugger)`, I got the exact same error caused by Chrome.
![image](https://github.com/odoo/odoo/assets/95914584/e2f20fa3-fc57-4632-adb0-287ddedd465c)

So it doesn't actually seem possible.